### ISSUE TITLE
feeds not loading 

### ISSUE THREAD

**Issue by Achef** [PrivacyVandaag/PrivacyVandaag#4]
Hello,
If feeds URLS contain old date, it will work only first time fetch rss then after it will not update 
because of old dates 

Examples of feeds dates
June 29, 2017, 9:29 AM
June 29, 2017, 9:29 AM
June 9, 2017, 10:33 AM
June 9, 2017, 10:33 AM
June 2, 2017, 9:29 AM
#June 29, 2017, 9:29 AM
June 7, 2017, 10:33 AM
June 2, 2017, 10:33 AM
June 29, 2017, 9:29 AM
June 29, 2017, 9:29 AM
June 29, 2017, 9:29 AM
June 29, 2017, 9:29 AM
June 29, 2017, 9:29 AM

I hope you can find solution for this issue.

**Comment by PrivacyVandaag** [comment]
That is correct. The app is designed that way. It presumes that all feeds are ordered chronologically. It presumes the most recent item is at the top of the list and the oldest item is at the bottom. Almost all feeds are built that way. 

The first time the whole feed is read. Thereafter, the feed is read until the app encounters an item with a publication date older than the last refresh. The next items of the feed are considered to be old and already fetched and are not analyzed for a second time. This is an efficient way of reading the feeds and saves battery power.

If you want to changes this, go to the RSSAtomParser.java and remove 
`                if (mItemPubDate.after(mLastUpdateDate) && mItemPubDate.after(mKeepDateBorder)) { ...}`
at line 646. Each time a feed is refreshed, the whole feed will be analyzed again.

**Comment by Achef** [comment]
I remove 
if (mItemPubDate.after(mLastUpdateDate) && mItemPubDate.after(mKeepDateBorder)) {
app getting crash!
How can I remove it without getting errors ?

**Comment by Achef** [comment]
Hello PrivacyVandaag,
I have been removed if (mItemPubDate.after(mLastUpdateDate) && mItemPubDate.after(mKeepDateBorder)) { ...} from RSSAtomParser.java
and still not working.. 
Nothing change.. 
is there any why to solve this issue?

__________________
I have been sent an email include feed link.Please check your email.

Thank you for your attention to this matter.

**Comment by PrivacyVandaag** [comment]
Dear Achef,
I am afraid I do not know what the issue with your implementation is. As I have written in my comment above, the reading of the RSS feed is stopped as soon as an already read item is encountered. 

So the solution for your problem would be to let the app read the entire rss feed each time and not checking if an item has already been read a previous time. This is not recommended as the app will use more battery power, but if it is necessary, one can remove the part where the app checks if an item has already been read. That is the line you mention.

Notice that items that are already in the database may be analyzed again, but are not always stored again or updated in the database! If a item has already been read, the app will update some information if that is necessary, but it can also skip the item and go to the next. This is done in the database query at the end of the function endElement() in the RssAtomParser class.

By the way, we have uploaded a major update of the app to make it compatible with android 8. Be aware that througout the app changes are made in the classes. RssAtomParser is much simpler now and a lot of comments are added to make the process easier to understand. The RSSAtomParser will probably not work in older versions of the app, but maybe it helps you to understand better the process that is going on there.
